nd4j/nd4j-common/src/main/java/org/nd4j/validation/Nd4jCommonValidator.java

```diff
@@ -0,0 +1,218 @@
+package org.nd4j.validation;
+
+import lombok.NonNull;
+import org.apache.commons.io.FileUtils;
+import org.nd4j.shade.jackson.databind.JavaType;
+import org.nd4j.shade.jackson.databind.ObjectMapper;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+public class Nd4jCommonValidator {
+
+    private Nd4jCommonValidator(){ }
```

Comment:
Total personal preference, but just curious, any reason we don't adapt lombok in the code base?

Reply:
We use Lombok extensively. But the equivalent annotation is actually longer in this particular case :)
`@NoArgsConstructor(access = AccessLevel.PRIVATE)`